src/posix/platform/platform-posix.h

```diff
@@ -84,6 +84,13 @@ enum
     OT_EVENT_DATA_MAX_SIZE          = 1024,
 };
 
+enum CoprocessorMode
+{
+    UNKNOWN = 0,
+    RCP     = 1,
+    NCP     = 2,
```

Comment:
Should we add prefix `OT_COPROCESSOR_MODE` before the definition similar with the [otRadioState](https://github.com/openthread/openthread/blob/main/include/openthread/platform/radio.h#L373)?